File: render_perf.js
const puppeteer = require('puppeteer');
const path = require('path');

(async () => {
    const browser = await puppeteer.launch({ headless: "new" });
    const page = await browser.newPage();
    
    // Set viewport
    await page.setViewport({ width: 1440, height: 1080, deviceScaleFactor: 2 });
    
    // Convert absolute path to file:// URL properly
    const filePath = path.resolve('dashboard_performance.html');
    const fileUrl = 'file://' + filePath;
    console.log("Navigating to", fileUrl);
    
    await page.goto(fileUrl, { waitUntil: 'networkidle0' });
    
    const screenshotPath = '/Users/andresmcmahon/.gemini/antigravity/brain/f218c95f-9262-4d7f-8524-7ffec3215612/perf_redesign_test_' + Date.now() + '.webp';
    await page.screenshot({ path: screenshotPath, type: 'webp', quality: 90 });
    
    console.log("Saved performance dashboard screenshot to", screenshotPath);
    await browser.close();
})();

File: render_perf.py
from selenium import webdriver
from selenium.webdriver.chrome.options import Options
import os
import time

chrome_options = Options()
chrome_options.add_argument("--headless")
chrome_options.add_argument("--window-size=1440,1080")

# Setup driver (assuming selenium can find chrome)
driver = webdriver.Chrome(options=chrome_options)

file_path = f"file://{os.path.abspath('dashboard_performance.html')}"
driver.get(file_path)

time.sleep(2) # Wait for animation renders

out_file = f"/Users/andresmcmahon/.gemini/antigravity/brain/f218c95f-9262-4d7f-8524-7ffec3215612/perf_redesign_{int(time.time())}.png"
driver.save_screenshot(out_file)
print(f"Screenshot saved to {out_file}")

driver.quit()
